docs/image.rst

```diff
@@ -0,0 +1,108 @@
+Image
+*****
+
+.. py:module:: microbit
+
+Classes
+=======
+
+.. py:class:: Image(width_or_string=None, height=None, buffer=None)
+
+    If ``width_or_string`` is a string, creates an image from that string.
+
+    If it's an integer, a height has to also be specified, and it creates
+    an image with ``width_or_string`` columns and ``height`` rows. Optionally
+    ``buffer`` can be an array of ``width_or_string``×``height`` integers to
+    initialize the image.
+
+
+    .. py:method:: width()
+
+        Return the number of columns in the image.
+
+
+    .. py:method:: height()
+
+        Return the numbers of rows in the image.
+
+
+    .. py:method:: set_pixel(x, y, value)
+
+        Set the brightness of the pixel at column ``x`` and row ``y`` to the
+        ``value``, which has to be between 0 (dark) and 9 (bright).
+
+
+    .. py:method:: get_pixel(x, y)
+
+        Return the brightness of pixel at column ``x`` and row ``y`` as an
+        integer between 0 and 9.
+
+
+    .. py:method:: shift_left(n)
+
+        Return a new image created by shifting the picture left by ``n``
+        columns.
+
+
+    .. py:method:: shift_right(n)
+
+        Same as ``image.shift_left(-n)``.
+
+
+    .. py:method:: shift_up(n)
+
+        Return a new image created by shifting the picture up by ``n`` rows.
+
+
+    .. py:method:: shift_down(n)
+
+        Same as ``image.shift_up(-n)``.
+
+
+Attributes
+==========
+
+The ``Image`` class also has the following instances of itself included as its
+attributes:
```

Comment:
IMO, not only sane, but reasonably sensible  for immutable singleton objects.
My concern with the ever increasing list of images is that they drown out the methods on Image when tabbing for auto-completion.
Moving them to an `images` module would address both of our concerns.